### PR TITLE
Regex routing for Organization in PageBreadCrumbs.vue

### LINKED ISSUE #920: Implement page checks that rely on UUIDs
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Till now the page check mechanisms we were using were based off of route parameters that just check for `organizations`, `groups`, etc, but not necessarily the full page route. Now that we're moving forward with hydration, we're now able to check for URLs based on the full route including UUIDs for the given pages. Such changes should be made to the page check mechanisms for components like the breadcrumbs and sidebar :)

### Contribution

I'll be working on this soon! 😊

### PR DESCRIPTION
**Pull request by hkkubais**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
- [] I have not run the tests for the backend and frontend depending on what's needed for my changes as described in the [testing section of the contributing guide](CONTRIBUTING.md#testing)

---

### Description

This pull request enhances the route check mechanisms in PageBreadcrumbs.vue by introducing a new organizationRegex to validate routes with the formats /organizations/{UUID} and optionally /organizations/{UUID}/about. This update ensures precise page type determination and improves the logic for fetching organization details by ID from the organizationStore.

Additionally, the regex logic now includes support for the base URL and an optional ISO route. This enhancement makes the implementation more robust by dynamically handling routes depending on the environment (e.g., localhost:3000 for development or the production URL) and optional language prefixes like /en. The assertion on the URL now ensures it adheres to a specific shape while maintaining flexibility for localized and non-localized routes.
Testing for the backend and frontend has not yet been conducted.

### Related issue

- partially resolves #920

### CODE REVIEW

File: frontend/components/page/PageBreadcrumbs.vue [hunk 1]
```diff
@@ -96,13 +96,9 @@ const eventStore = useEventStore();
 let organization: Organization;
 let group: Group;
 let event: Event;
+const organizationRegex = /^\/organizations\/[0-9a-fA-F-]+(\/about)?$/;
```
Comment (cquinn540) [inline frontend/components/page/PageBreadcrumbs.vue:99]: Shouldn't we do the same thing for events?

Comment (hkkubais) [reply frontend/components/page/PageBreadcrumbs.vue:99]: Yeah, my teammate is working on that, however, we just wanted to check if we're on the right track.

**Comment by cquinn540** [comment]
You can test this with the component tests.

They can [render the app with the route argument](https://nuxt.com/docs/getting-started/testing#rendersuspended) set to a route with fake organization id, and then mock the api call to `/entities/organizations/${id}` with a fake organization and see if the name of their fake org shows up in the breadcrumbs.

Then also test a route like create route to make sure the breadcrumbs still work with that too.

**Comment by cquinn540** [comment]
Oh damn yeah it's going to have to be tested by the component tests because there's a bug keeping the breadcrumbs from being displayed on the necessary pages:
Screenshot 2024-12-10 at 2 44 14 PM

**Review by andrewtavis — APPROVED** [review]
Thanks for the initial contribution here, @hkkubais! We'll go through and make any needed changes to fix formatting or other issues as maintainers were not given edit rights to the PR. Appreciate your communication on this!

**Pull request merged**